community/blog/2023-01 W3C service-to-service correlation in newest Arcus releases/README.md

```diff
@@ -0,0 +1,99 @@
+# W3C service-to-service correlation in newest Arcus releases
+One of the biggest changes in the recent Arcus releases is the W3C correlation support on all of our components. Read further to learn how it can improve your application.
+
+## What has changed?
+The service-to-service correlation feature was already supported before we looked into W3C. Service-to-service correlation provides a way to link all the send-out telemetry of your application's services into a single transaction diagram. This helps pinpoint the cause of service failures, performance bottlenecks, and gives you a clear overview of all the resources your project uses. In Azure Application Insights, this translates to [transaction diagnostics](https://learn.microsoft.com/en-us/azure/azure-monitor/app/transaction-diagnostics) and an [application map](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-map?tabs=net).
+
+So, what has changed? Previously, we were using [hierarchical](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/HierarchicalRequestId.md) correlation, which uses `Request-Id` to establish a link between 'parent' (sender) and 'child' (receiver). This ID could be sent via an HTTP header or an Azure Service Bus application property or any other system that has a sender/receiver relationship. There were two problems with this approach, 1. the hierarchical correlation system is deprecated, and 2. it is not Microsoft's default correlation format. This made us switch to the W3C correlation system. The biggest changes are internal. Though, this format uses the `traceparent` value instead to pass along sender/receiver services. 
+
+![W3C HTTP correlation diagram](media/w3c-api-diagram.png)
+
+## Migrate from hierarchical to W3C correlation
+The W3C update affects three packages. Make sure that you update to at least these versions: [Arcus Web API v1.7](https://webapi.arcus-azure.net/), [Arcus Messaging v1.4](https://messaging.arcus-azure.net/), and [Arcus Observability v2.7](https://observability.arcus-azure.net/).
+
+🚩 Note that you do not need a Messaging package in an API application or the way around. These are just the minimum versions you'll need in any one of your applications using Arcus.
+
+Now comes the fun part. All you have to do in any of your applications to have seamless integration with Microsoft is add/change these lines in your `Program`:
```

Comment:
If your API is part of a larger application, would you not also need to make sure that the component calling your API sends out the `traceparent` header? For example, currently I add the `Request-Id` header from APIM policies to calls to API Functions, that would need to be changed as well right?

Reply:
If configured correctly, the `HttpClient` will automatically, internally, add the `traceparent` header (just like it will add the `Diagnostic-Id` application property to Azure Service Bus messages), so it is enough to remove the Arcus HTTP dependency tracking, here.

Reply:
You mean when calling from the API right? I mean calling the API from another component 😉

Reply:
Ah, like that. Yeah, it would be wise to streamline the same correlation system across your services. This should be added here, you mean?

Reply:
I think It would be good to mention this here, the text now suggests that you only need to change a couple of minor lines in the code. But that does not take into account that you need to change this across your services.

Reply:
Ok! 👍 Added a warning message here:  https://github.com/arcus-azure/arcus/pull/296/commits/ba571e16881b7b5448603116aecd561447246cbf